sdk/java-sdk-protobuf/src/main/resources/reference.conf

```diff
@@ -93,4 +93,11 @@ kalix {
       }
     }
   }
+
+  telemetry {
+     tracing {
+        collector-endpoint = "http://localhost:4317"
+        collector-endpoint =  ${?COLLECTOR_ENDPOINT}
+     }
+  }
```

Comment:
So, this is always enabled against a local endpoint when user is running tests, integration tests, locally etc? Shouldn't it rather be disabled by default and then opt in? Will the user be running something on port 4317?

Reply:
Right. Did opt-in in the proxy but forgot to do it here.
The UF will be pointing to an OpenTelemetry container sidecar

Reply:
Actually, I remember now that I thought that one shouldn't enable or disable the tracing in the SDK. It will propagate tracing if it receives traces from the proxy otherwise won't do nothing.

Reply:
see [here](https://github.com/lightbend/kalix-jvm-sdk/pull/1763/files/a59c5f1c108c310537b5817d4f9669387f716cf5#diff-0eafcaa0a75ac0670d8a260a9d73a07dd2e7aba36d82e183e6b67762a360a7e4R89)  please

Reply:
That means it should lazily create the Telemetry instances the first time it sees that they are needed rather than up front like it is currently done.

Reply:
I see. For that I would need a mutable Map (sort of a cache) because we don't know if tracing is coming from the proxy until we process the first command. We find one way or another inside the metadata. Hence I need to delay the construction until then. The problem I see is that such a cache  won't be thread safe. WDYT?

I looked at http://eed3si9n.com/learning-scalaz/Memo.html but I'm not sure we want this either. 

That said I think this optimization may not be necessary. Even if we have tenths of services I don't think it will be a to create that amount of Telemetry objects.

Reply:
I can use a [TrieMap](https://www.scala-lang.org/api/current/scala/collection/concurrent/TrieMap.html) in case you still think is worth to optimize this creation.

Reply:
Maybe better a `java.util.concurrent.ConcurrentHashMap`

Reply:
I'm implementing it and I think it will perform worse when checking every request if the Telemetry has been created on not in a `ConcurrentHashMap`. Not only it need to do the check first (in the best case scenario, i.e. the Telemetry instance exists already), it will sync (block) with any other thread that is accessing that `ConcurrentHashMap`. 

Does that makes sense?

Reply:
Just to sum up what we discussed latest, it seems we want one OpenTelemetry sdk instance per component, but we can probably eager create those optionally in `...Impl` constructor so that no concurrency is needed, and only if tracing is enabled/needed for a given SDK process.